src/internal/operators/share.ts

```diff
@@ -1,14 +1,51 @@
+/** @prettier */
 import { Observable } from '../Observable';
-import { multicast } from './multicast';
-import { refCount } from './refCount';
 import { Subject } from '../Subject';
 
-import { MonoTypeOperatorFunction } from '../types';
+import { MonoTypeOperatorFunction, OperatorFunction } from '../types';
+import { Subscription } from '../Subscription';
+import { from } from '../observable/from';
+import { lift } from '../util/lift';
+import { Subscriber } from '../Subscriber';
 
-function shareSubjectFactory() {
-  return new Subject<any>();
+interface ShareOptions<T, R> {
```

Comment:
(very minor) nitpick: In [other parts of the API](https://github.com/ReactiveX/rxjs/blob/8ef4fee2b5f7fb68426661ffcc47b50ab81b145e/src/internal/operators/throttle.ts#L9), these types/parameters are referred to as 'config'. I think it should be one or the other and should be consistent throughout the API, but - again - this is very minor.